### DISCUSSION TITLE
Does AGP provide dependencies?

### DISCUSSION THREAD

**Discussion by ghost** [allure-framework/allure2#2729] [Questions & Support] [answered]
I noticed that the Allure plugin for Gradle provides Allure dependencies.

With the default configuration of AGP 2.12.0 I can see in my rpoject:
![image](https://github.com/user-attachments/assets/bab31b9d-3a22-4f2d-864d-24613532b536)

Does this mean I don't need to explicitly add dependencies?
And if so, what to do if, for example, AssertJ integration is needed?

**Comment by baev** [comment, marked as answer]
The Allure Gradle plugin can configure dependencies for you automatically based on the dependencies it detects in your project. For example, in case of `testng` dependency present, it will automatically add `allure-testng` dependency and so on. 

AspectJ can be configured automatically as well:

```kts
allure {
    adapter {
        aspectjVersion.set("1.9.22.1")
        aspectjWeaver.set(true)
    }
}
```

for more configuration options please see https://github.com/allure-framework/allure-gradle
